snapcraft/_options.py

```diff
@@ -0,0 +1,41 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+class _Project:
+
+    @property
+    def use_geoip(self):
+        return self.__use_geoip
+
+    @use_geoip.setter
+    def use_geoip(self, setting):
+        self.__use_geoip = setting
```

Comment:
El miércoles, 6 de abril de 2016 14h'52:17 ART, Kyle Fazzari 
<notifications@github.com> escribió:
>> +# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
>> +# GNU General Public License for more details.
>> +#
>> +# You should have received a copy of the GNU General Public License
>> +# along with this program.  If not, see <http://www.gnu.org/licenses/>.
>> +
>> +
>> +class _Project:
>> +
>> +    @property
>> +    def use_geoip(self):
>> +        return self.__use_geoip
>> +
>> +    @use_geoip.setter
>> +    def use_geoip(self, setting):
>> +        self.__use_geoip = setting
> 
> What is the benefit of this encapsulation?

By doing it this way we can change the implementation internally without 
worrying about the external interface. Also no one would be able to reach 
this.


-- 
Enviado con Dekko desde mi dispositivo Ubuntu